src/nvim/msgpack_rpc/channel.c

```diff
@@ -41,7 +43,8 @@
 typedef enum {
   kChannelTypeSocket,
   kChannelTypeProc,
-  kChannelTypeStdio
+  kChannelTypeStdio,
+  kChannelTypeInternal
 } ChannelType;
```

Comment:
Missing case for new enum element in some `switch(channel->type)`. The `default` case for a `switch` over an `enum` should be removed, so that the compiler can warn us.